File: annotationDemo/src/main/java/com/demo/my/myapplication/MainActivity.java
package com.demo.my.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.demo.my.annotation_lib.version;


@version(ver = 2, name = "MainActivity")
public class MainActivity extends AppCompatActivity {

    private TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tv = findViewById(R.id.tv);
        tv.setText(ProviderInit.init());
    }
}

File: kotlindemo/src/main/java/com/demo/my/kotlindemo/MainActivity.kt
package com.demo.my.kotlindemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.demo.my.kotlindemo.bean.Person

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init()
    }

    private fun init() {
        //创建对象
        val testPerson = Person("Alex", 16, null)

    }

}
